test/router_test.exs

```diff
@@ -224,6 +250,32 @@ defmodule BeeBee.RouterTest do
       assert conn.status == 204
     end
 
+    test "blocks unauthenticated update" do
+      conn =
+        :put
+        |> conn("/_update", %{"url" => "https://github.com", "short_tag" => "grape"})
+        |> Router.call([])
+
+      assert conn.status == 401
+    end
+
+    test "allows authenticated update" do
+      encoded_credentials = Plug.BasicAuth.encode_basic_auth("user", "password")
+
+      :post
+      |> conn("/_add", %{"url" => "https://github.com", "short_tag" => "yuh"})
+      |> Plug.Conn.put_req_header("authorization", encoded_credentials)
+      |> Router.call([])
+
+      conn =
+        :put
+        |> conn("/_update", %{"url" => "https://github.com", "short_tag" => "yuh"})
```

Comment:
It would probably be more useful to use a different url for the update and test to see that it gets updated.